src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1750,7 +1750,8 @@ public override void VisitMarkupTagHelperElement(MarkupTagHelperElementSyntax no
                 TagName = tagName,
                 TagMode = info.TagMode,
                 Source = BuildSourceSpanFromNode(node),
-                TagHelpers = info.BindingResult.Descriptors
+                TagHelpers = info.BindingResult.Descriptors,
+                StartTagSpan = node.StartTag.Name.GetSourceSpan(SourceDocument)
```

Comment:
`TagHelperIntermediateNode` is created in two places in `DefaultRazorInteremediateNodeLoweringPhase`: here in `ComponentFileKindVisitor.VisitMarkupTagHelperElement(...)` and on line 1031 in the `LegacyFileKindVisitor.VisitMarkupTagHelperElement(...)`. Should this property be set in the `LegacyFileKindVisitor` as well? If not, consider adding a comment to explain why.